### DISCUSSION TITLE
Polygon example

### DISCUSSION THREAD

**Discussion by meniv-nl** [clickbar/laravel-magellan#82] [Q&A] [answered]
Great library! 
One question; could you add a example how to add a polygon to the db? (lineStrings format?)

**Comment by saibotk** [comment, marked as answer]
Hey 👋 

Sorry for the late reply, here is a simple example for adding a Polygon to the Ship model (has a polygon field called `poly`)

```php
use Clickbar\Magellan\Data\Geometries\LineString;
use Clickbar\Magellan\Data\Geometries\Point;
use Clickbar\Magellan\Data\Geometries\Polygon;

// .... 

$ship->poly = Polygon::make([
    LineString::make([
        Point::makeGeodetic(1, 1),
        Point::makeGeodetic(2, 2),
        Point::makeGeodetic(3, 3),
        Point::makeGeodetic(1, 1),
    ]),
]);

$ship->save();
```

Hope this helps!

**Reply by mikecobb-io** [reply]
How would I best retrieve the above polygon data to use in Azure Maps which requires polygon data as an array of points, like so:

`new atlas.data.Polygon([[[-73.98235, 40.76799],[-73.95785, 40.80045],[-73.94928, 40.7968],[-73.97317, 40.76437],[-73.98235, 40.76799]]])`

Thanks in advance

**Reply by saibotk** [reply]
The structure seems to be the same you can just access the linestrings on a Polygon and its points and map the data to the array structure.

Or depending on the usecase you can skip the cast to the polygon and instead try to do that in sql directly. Though i am not at my pc atm.

**Reply by mikecobb-io** [reply]
Thanks. I'm looking for a code snippet that will query and get just the polygon coordinates as I can't write/find a query that just returns the geometry coordinates without them being nested in the features[0][geometry][0][coordinates][0] array, or is that the expected result. Any pointers much appreciated

**Reply by saibotk** [reply]
I think this might be useful as it returns all points individually

https://postgis.net/docs/ST_DumpPoints.html
